CONTRIBUTING.md

```diff
@@ -221,6 +232,48 @@ Invoked when the custom element is disconnected from the document's DOM.
 
 The render loop makes sure that upon each [`attributeChangedCallback()`](#attributechangedcallbackname-oldvalue-newvalue) invocation or any observed [property `setter()`](#property-setter) invocation that the flattened DOM is recomputed and that [`willRenderCallback()`](#willrendercallbackinitial) and [`didRenderCallback()`](#didrendercallbackinitial) lifecycle hooks are called respectively.
 
+## Integration
+
+The goal is that custom elements can be shared across frameworks and libraries like Angular, React, Vue, you name it. To ease this process we provide generic wrapper functions.
+
+### `withReact()`
+
+To turn any custom element into a working React Component, you just need to follow these steps:
+
+1. `import` React
+2. `import` withReact
+3. `import` any web components you need
+4. bind `withReact` to you `import`ed React version
+5. wrap all your needed web components
+   - and may pass optional options for type of component or event init options
+6. use them like regular React components in your app
+
+   **Note:** events work similar to React's standard events, but each web components could trigger custom events like `axa-click` - camelcased and `on`-prefixed in React such as `onAxaClick={yourEventHandler}`. Make sure to check them out at the web-components documentation itself!
+
+```js
+// import your dependencies - 1, 2, and 3
+import React from 'react';
+import withReact from '@axa-ch/patterns-library/src/js/with-react';
+import AXAButton from '@axa-ch/patterns-library/dist/components/m-button';
+
+// 4. bind withReact to your imported Reat version
+const withReactBound = withReact(React)
```

Comment:
Why didn't you use a peerDependency to React here?

Reply:
super valid point, this would simply the API by avoiding partial application 😃 
Though I would like to know if `peerDependencies` can be optional or limited to specific imports.
I wouldn't want to create a separate npm package just for `withReact`.
Will do some research on this.

Reply:
Either add it to peerDependencies and have a warning if it is not installed/in wrong version OR do not add it to dependencies at all. Will work same as peerDependencies. Both will yield a runtime/webpack exception when react is not in node_modules

Reply:
Yeah there are also [`optionalDependencies`](https://docs.npmjs.com/files/package.json#optionaldependencies) 🎉

Reply:
Only downside is:
> Entries in optionalDependencies will override entries of the same name in dependencies, so it's usually best to only put in one place.

And we need React in the patterns-lib app being installed for the demos...

Reply:
peerDependencies will yieald a warning during `npm i`/`yarn` but no error/abort

Reply:
@sventschui Super awesome. Thanks for your feedback. Just pushed it:)